### ISSUE TITLE
Instructions how to compile this with Android Studio

### ISSUE THREAD

**Issue by CaptainMJx** [aws/aws-sdk-java-v2#2751]
Hi! Are there any instructions available for how to compile this for Android? As I understand the Mobile SDK for Android is being replaced by Amplify so the only option left then is to use this Java SDK.  Amplify looks to be more of a simplified help for app developers to easily create backend support functions, not the other way around and it lacks a lot of functionality to communicate with existing aws backends like shadows, mqtt topics etc... So if there is any guidance on how to build this in Android Studio that would be very much appreciated.

Thanks,
Marcus

**Comment by debora-ito** [comment]
We don't have the expertise of working in Android environments, unfortunately. In those cases we would recommend the Mobile SDK, and now we recommend using Amplify. 

Besides support for shadows and mqtt, is there anything else you would like to see in Amplify? Which services are you going to interact with, mostly?

**Comment by CaptainMJx** [comment]
Hi Debora,

 
Thanks for your email. I think I can manage the Android specifics myself (though it would have been great with some assistance). In any case, I do need help on how to use Cognito with AWS Java SDK v2, for normal username password authentication, is there any example code available for this? I think this would be the same or at least similar use case as when using a java desktop application.

 
Amplify does not seem very useful for controlling an existing infrastructure, this seems to me more like an app driven framework where the developer get the cloud infrastructure setup automatically by the Amplify framework to fit the app requirements, which is not what I am doing. 

 
In regards to features in Amplify, it would be the SDK of choice if I was able to interact with Device Shadows, do MQTT publish and subscribe, execute Lambda functions(this part may already be supported, not sure). And Finally, a way to configure existing cloud services in Amplify (or remove the need for that), currently it’s very hard and also hard to find information on how to setup Amplify to use existing infrastructure. So some way to use Amplify for situations where the cloud infra is already setup, instead of having Amplify creating all of it.

 
In short, it seems to me that Amplify is purpose built for general App development and not as an SDK to interact with an existing system running in the cloud. However, if I am wrong about this I am happy to have another look at Amplify.

 
Many thanks,

Marcus
